Feature/cde 39

afonsobspinto opened the pull request:
Closes https://metacell.atlassian.net/browse/CDE-39

- Adds optional getPairingSuggestions property to collection
- Implements example of getPairingSuggestions from interlex as explained [here](https://metacell.slack.com/archives/C05D8JSJ8ER/p1710856000721219).
- Creates usePairingSuggestions hook to manage pairing suggestions processing 
- Connects mapping tab: 
Upon CDE selection we check for pairing suggestions on each selected collection.
If we have suggestions we show the collapsible tooltip.
The user can selected any non-mapped column to map it to the suggestion.
When a suggestion is accepted we update the dataset mapping accordingly.
- Connects PairingSuggestions and PairingTooltip components (previously with mocked data)


Questions:
- Limit pairing suggestions? (we are fetching for at most 20 per collection and showing them all)
- Tooltip message?
![image](https://github.com/MetaCell/cde-mapper/assets/19196034/be6ec6bc-db76-41ce-bfc6-1540d1c15b38)

We might need some UI improvements.

https://github.com/MetaCell/cde-mapper/assets/19196034/fa3cf5a4-c54b-43cf-b40d-b65a71063fdb

afonsobspinto committed:
CDE-39 feat: Add fetch for related

afonsobspinto committed:
CDE-39 feat: Show pairing suggestions

afonsobspinto committed:
CDE-39 feat: Connect pairing suggestions

ddelpiano commented:
@afonsobspinto 
Limit the pairing suggestions - no
Tooltip message - In this section the CDE Mapper will present fellow CDEs that are meaningful and usually associated with the parent of the pairing suggestions.

afonsobspinto committed:
CDE-39 feat: Update pairing tooltip

afonsobspinto committed:
CDE-39 feat: Remove related query size

afonsobspinto commented:
> @afonsobspinto Limit the pairing suggestions - no Tooltip message - In this section the CDE Mapper will present fellow CDEs that are meaningful and usually associated with the parent of the pairing suggestions.

Updated tooltip:
![image](https://github.com/MetaCell/cde-mapper/assets/19196034/9d834f29-696b-447c-b4f1-c6ac39948e42)

I also removed the size:20 from the pairing suggestions query, so there's no limit on how many we can get (which in theory can be problematic, but maybe not in practice)
